### PR TITLE
Add project variable set relationship APIs

### PR DESCRIPTION
**Pull request by joekarl**
### What
Variable sets can be assigned to projects via API. This documents those APIs.

### Why
Documents an existing API.

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [X] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [X] Description links to related pull requests or issues, if any.

#### Content
- [ ] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [X] API documentation and the API Changelog have been updated. 
- [ ] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [ ] Pages with related content are updated and link to this content when appropriate.
- [ ] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [ ] New pages have metadata (page name and description) at the top.
- [ ] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [ ] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [ ] UI elements (button names, page names, etc.) are bolded.
- [ ] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/api-docs/variable-sets.mdx [hunk 1]
```diff
@@ -256,120 +292,203 @@ curl \
...
 
-List all variable sets for a workspace.
+`GET projects/:project_id/varsets`
+
+| Parameter     | Description    |
+|---------------|----------------|
+| `:project_id` | The project ID |
+
+List all variable sets for a workspace. This includes global variable sets from the workspace's organization and variable sets
+attached to the workspace's parent project.
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/variable-sets.mdx:385]: From what I've seen, we usually say workspaces are "contained within" projects:
```suggestion
attached to the project this workspace is contained within.
```
Please feel free to push back if you don't like this phrasing!

File: website/docs/cloud-docs/api-docs/variable-sets.mdx [hunk 2]
```diff
@@ -687,6 +806,103 @@ curl \
   https://app.terraform.io/api/v2/varsets/varset-kjkN545LH2Sfercv/relationships/workspaces
 ```
 
+## Apply Variable Set to Projects
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/variable-sets.mdx:809]: How would you feel about moving this section above "Apply Variable Set to Workspaces"? In order of specificity, it feels it make sense to tell folks how to assign to projects _before_ workspaces.

File: website/docs/cloud-docs/api-docs/variable-sets.mdx [hunk 3]
```diff
@@ -687,6 +806,103 @@ curl \
...
+
+Accepts a list of projects to add the variable set to. When the variable set is applied to a project, all workspaces in the project will have the variable set applied to them.
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/variable-sets.mdx:811]:
```suggestion
Accepts a list of projects to add the variable set to. When you apply a variable set to a project, all the workspaces in that project will have the variable set applied to them.
```

**Review by rkoron007 — COMMENTED** [review]
I left some suggestions for style and reformatting, but _please_ feel free to push back on anything that doesn't make sense to you!

File: website/docs/cloud-docs/api-docs/variable-sets.mdx (hunk 2 above)
Comment (joekarl) [reply website/docs/cloud-docs/api-docs/variable-sets.mdx:809]: That makes sense, I didn't know if there was a preference on ordering.

**Review by rkoron007 — APPROVED** [review]
This looks good to me! ✨

**Pull request merged**